[Fabric] Server failed to install (error code 400)

Hello,

When trying to install Fabric version 1.20.1 with the Fabric template, you get the following error:

```
Downloading required files
service FabricService{meta='https://meta.fabricmc.net/', maven='https://maven.fabricmc.net/'} failed: java.io.IOException: HTTP request to https://meta.fabricmc.net/v2/versions/loader/1.20.1/latest/server/json failed: 400
service FabricService{meta='https://meta2.fabricmc.net/', maven='https://maven2.fabricmc.net/'} failed: java.io.IOException: HTTP request to https://meta2.fabricmc.net/v2/versions/loader/1.20.1/latest/server/json failed: 400
service FabricService{meta='https://meta3.fabricmc.net/', maven='https://maven3.fabricmc.net/'} failed: java.net.UnknownHostException: meta3.fabricmc.net
Exception in thread "main" java.lang.RuntimeException: Failed to install Server
	at net.fabricmc.installer.Main.main(Main.java:89)
Caused by: java.io.IOException: HTTP request to https://meta.fabricmc.net/v2/versions/loader/1.20.1/latest/server/json failed: 400
	at net.fabricmc.installer.util.Utils.openUrl(Utils.java:148)
	at net.fabricmc.installer.util.Utils.readString(Utils.java:90)
	at net.fabricmc.installer.util.FabricService.lambda$queryMetaJson$0(FabricService.java:36)
	at net.fabricmc.installer.util.FabricService.invokeWithFallbacks(FabricService.java:79)
	at net.fabricmc.installer.util.FabricService.queryMetaJson(FabricService.java:36)
	at net.fabricmc.installer.server.ServerInstaller.install(ServerInstaller.java:83)
	at net.fabricmc.installer.server.ServerInstaller.install(ServerInstaller.java:66)
	at net.fabricmc.installer.server.ServerHandler.installCli(ServerHandler.java:78)
	at net.fabricmc.installer.Main.main(Main.java:87)
	Suppressed: java.io.IOException: HTTP request to https://meta2.fabricmc.net/v2/versions/loader/1.20.1/latest/server/json failed: 400
		... 9 more
	Suppressed: java.net.UnknownHostException: meta3.fabricmc.net
		at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:572)
		at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
		at java.base/java.net.Socket.connect(Socket.java:633)
		at java.base/sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:304)
		at java.base/sun.net.NetworkClient.doConnect(NetworkClient.java:178)
		at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:533)
		at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:638)
		at java.base/sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:266)
		at java.base/sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:380)
		at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:193)
		at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1242)
		at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1128)
		at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:179)
		at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.connect(HttpsURLConnectionImpl.java:142)
		at net.fabricmc.installer.util.Utils.openUrl(Utils.java:145)
		... 8 more
```

Thank you,
Polarian

Same here, bump
```
[DAEMON] Installing server
DAEMON  Downloading Java 17
DAEMON  Downloading metadata from https://meta.fabricmc.net/v2/versions/installer
DAEMON  Downloading: https://maven.fabricmc.net/net/fabricmc/fabric-installer/0.11.2/fabric-installer-0.11.2.jar
DAEMON  Executing: java17 -jar fabric-installer.jar server -mcversion 1.19.2 -loader latest -downloadMinecraft -noprofile
DAEMON  Starting process: /var/lib/pufferpanel/binaries/java17 -jar fabric-installer.jar server -mcversion 1.19.2 -loader latest -downloadMinecraft -noprofile
Loading Fabric Installer: 0.11.2
Installing Fabric Loader latest(1.19.2) on the server
Downloading required files
service FabricService{meta='https://meta.fabricmc.net/', maven='https://maven.fabricmc.net/'} failed: java.io.IOException: HTTP request to https://meta.fabricmc.net/v2/versions/loader/1.19.2/latest/server/json failed: 400
service FabricService{meta='https://meta2.fabricmc.net/', maven='https://maven2.fabricmc.net/'} failed: java.io.IOException: HTTP request to https://meta2.fabricmc.net/v2/versions/loader/1.19.2/latest/server/json failed: 400
service FabricService{meta='https://meta3.fabricmc.net/', maven='https://maven3.fabricmc.net/'} failed: java.net.UnknownHostException: meta3.fabricmc.net
Exception in thread "main" java.lang.RuntimeException: Failed to install Server
	at net.fabricmc.installer.Main.main(Main.java:89)
Caused by: java.io.IOException: HTTP request to https://meta.fabricmc.net/v2/versions/loader/1.19.2/latest/server/json failed: 400
	at net.fabricmc.installer.util.Utils.readString(Utils.java:90)
	at net.fabricmc.installer.util.Utils.openUrl(Utils.java:148)
	at net.fabricmc.installer.util.FabricService.lambda$queryMetaJson$0(FabricService.java:36)
	at net.fabricmc.installer.util.FabricService.invokeWithFallbacks(FabricService.java:79)
	at net.fabricmc.installer.util.FabricService.queryMetaJson(FabricService.java:36)
	at net.fabricmc.installer.server.ServerInstaller.install(ServerInstaller.java:83)
	at net.fabricmc.installer.server.ServerInstaller.install(ServerInstaller.java:66)
	at net.fabricmc.installer.server.ServerHandler.installCli(ServerHandler.java:78)
		... 9 more
	Suppressed: java.io.IOException: HTTP request to https://meta2.fabricmc.net/v2/versions/loader/1.19.2/latest/server/json failed: 400
	at net.fabricmc.installer.Main.main(Main.java:87)
		at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:572)
	Suppressed: java.net.UnknownHostException: meta3.fabricmc.net
		at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
		at java.base/java.net.Socket.connect(Socket.java:633)
		at java.base/sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:304)
		at java.base/sun.net.NetworkClient.doConnect(NetworkClient.java:178)
		at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:533)
		at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:638)
		at java.base/sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:266)
		at java.base/sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:380)
		at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:193)
		at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1242)
		at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1128)
		at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:179)
		at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.connect(HttpsURLConnectionImpl.java:142)
		at net.fabricmc.installer.util.Utils.openUrl(Utils.java:145)
		... 8 more
DAEMON  Writing some data to file: server.properties
DAEMON  Writing some data to file: eula.txt
DAEMON  Creating directory: mods
DAEMON  Creating directory: installer
DAEMON  Moving file from fabric-installer.jar to installer/fabric-installer.jar
DAEMON  Server installed
```

I looked at https://meta.fabricmc.net/v2/versions/loader/ and saw that versions are distributed differently than what the installer installs them like, 1.19.2 (or in OP's case 1.20.1) doesn't exist

Edit: upon looking deeper /latest doesn't seem to work, https://meta.fabricmc.net/v2/versions/loader/1.19.2/latest

Strange that there's no 404 error but we get 400 so I'm not sure what's going on here

2ndEdit: https://meta.fabricmc.net/v2/versions/loader/1.20.1/0.1.0.48 works instead of https://meta.fabricmc.net/v2/versions/loader/1.20.1/latest
Not sure why they'd not make latest but looks like you have to fetch 1st position in the list at https://meta.fabricmc.net/v2/versions/loader/1.20.1/ and then pick the first (cause in api they say 1st is always latest) or input a version.

**Work around right now is setting fabric version to 0.14.23 (latest) (from experience tho .22 is more stable)**

--- a/minecraft-fabric/minecraft-fabric-docker.json
+++ b/minecraft-fabric/minecraft-fabric-docker.json
@@ -14,7 +14,7 @@
       "desc": "Version of Minecraft to install (Oldest fabric version is 1.14)",
       "display": "GAME VERSION",
       "required": true,
-      "value": "1.20.1"
+      "value": "1.20.2"
     },
     "loader-version": {
       "type": "string",

--- a/minecraft-fabric/minecraft-fabric-docker.json
+++ b/minecraft-fabric/minecraft-fabric-docker.json
@@ -16,13 +16,6 @@
       "required": true,
       "value": "1.20.2"
     },
-    "loader-version": {
-      "type": "string",
-      "desc": "Version of Fabric Loader to install",
-      "display": "LOADER VERSION",
-      "required": true,
-      "value": "0.15.0"
-    },
     "ip": {
       "type": "string",
       "desc": "What IP to bind server to",
@@ -58,7 +51,7 @@
     },
     {
       "commands": [
-        "java -jar fabric-installer.jar server -mcversion ${game-version} -loader ${loader-version} -downloadMinecraft -noprofile"
+        "java -jar fabric-installer.jar server -mcversion ${game-version} -downloadMinecraft -noprofile"
       ],
       "type": "command"
     },

--- a/minecraft-fabric/minecraft-fabric-docker.json
+++ b/minecraft-fabric/minecraft-fabric-docker.json
@@ -21,7 +21,7 @@
       "desc": "Version of Fabric Loader to install",
       "display": "LOADER VERSION",
       "required": true,
-      "value": "latest"
+      "value": "0.15.0"
     },
     "ip": {
       "type": "string",

